File: test/Z.Core.Test/System.TimeSpan/TimeSpan.FromNow.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Z.Core.Test
{
    [TestClass]
    public class System_TimeSpan_FromNow
    {
        [TestMethod]
        public void FromNow()
        {
            // Type
            var @this = new TimeSpan(1, 0, 0, 0);

            // Examples
            DateTime value = @this.FromNow(); // return tomorrow.

            // Unit Test
            Assert.IsTrue(DateTime.Now.Subtract(value).Days <= -1);
        }
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.TimeSpan/TimeSpan.FromNow.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library


Public Module Extensions_607

	''' <summary>
	'''     A TimeSpan extension method that add the specified TimeSpan to the current DateTime.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <returns>The current DateTime with the specified TimeSpan added to it.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function FromNow(this As TimeSpan) As DateTime
		Return DateTime.Now.Add(this)
	End Function
End Module


